### PR TITLE
Migrate Min Hashing APIs to pylibcudf

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Apart of #15162 
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/libcudf/utilities/span.pxd [hunk 1]
```diff
@@ -7,3 +7,6 @@ cdef extern from "cudf/utilities/span.hpp" namespace "cudf" nogil:
     cdef cppclass host_span[T]:
         host_span() except +
         host_span(vector[T]) except +
+    cdef cppclass device_span[T]:
```
Comment (Matt711) [inline python/pylibcudf/pylibcudf/libcudf/utilities/span.pxd:10]: I think we prefer to use a `column_view` over a `device_span` (eg. we overload the `seeds` arguments in the min hashing functions to use a `column_view`. Should I delete `device_span`?

**Review by mroeschke — APPROVED** [review]
I don't have a strong opinion about the `device_span`. Since it's not a detail API, I think technically it should be exposed in pylibcudf

**Comment by vyasr** [comment]
I don't expect pylibcudf users to ever need the equivalent of a `device_span`. In C++ spans are useful as a way to create views on data owned by another object, but in pylibcudf Columns already have this ability baked in because they have to match Python's referencing semantics. The pylibcudf implementation in this PR demonstrates that since they only take Columns (you didn't bother to create a span overload). With that in mind, I don't think it's important to expose span-based libcudf APIs to pylibcudf if there is a column-based alternative.

**Pull request merged**